### DISCUSSION TITLE
InputPhoneComponent workaround

### DISCUSSION THREAD

**Discussion by Joshua-Ryan** [taiga-family/taiga-ui#1609] [Q&A] [answered]
I have a task where I need the tui-input-phone component to emit an object instead of just a string. In the past I have used a transformer token if it is supplied to the base class like in the InputDateComponent. Another time I have used a value accessor, however when I am writing a value accessor for the phone component onValueChange is never caught with a host listener. I am wondering if this is intended or somehow the component is missing the ability to alter the value before it gets to the form.

**Comment by waterplea** [comment, marked as answer]
Transformer, like in `InputDate`, is the way to go for such task. I see it is only implemented for date controls at the moment, please file an issue to be able to provide transformers for every control.
